Extensions/xkit_preferences.js

```diff
@@ -378,14 +382,36 @@ XKit.extensions.xkit_preferences = new Object({
 			var show_all = XKit.tools.get_setting("xkit_show_feature_updates","true") === "true";
 
 			var m_return = 0;
+			var show_information = JSON.parse(XKit.tools.get_setting("xkit_show_information","true"));
+			var show_extension_updates = JSON.parse(XKit.tools.get_setting("xkit_show_extension_updates","true"));
+			var show_weekly_news = JSON.parse(XKit.tools.get_setting("xkit_show_weekly_news","true"));
+			var show_developer_news = JSON.parse(XKit.tools.get_setting("xkit_show_developers","true"));
+			var show_firefox = true;
+			var show_chrome = true;
+			var show_safari = true;
+			if(XKit.browser().firefox) {
+				show_firefox = JSON.parse(XKit.tools.get_setting("xkit_show_firefox_news","true"));
+				show_chrome = JSON.parse(XKit.tools.get_setting("xkit_show_chrome_news","false"));
+				show_safari = JSON.parse(XKit.tools.get_setting("xkit_show_safari_news","false"));
+			} else if(XKit.browser().chrome) {
+				show_firefox = JSON.parse(XKit.tools.get_setting("xkit_show_firefox_news","false"));
+				show_chrome = JSON.parse(XKit.tools.get_setting("xkit_show_chrome_news","true"));
+				show_safari = JSON.parse(XKit.tools.get_setting("xkit_show_safari_news","false"));
+			} else if(XKit.browser().safari) {
+				show_firefox = JSON.parse(XKit.tools.get_setting("xkit_show_firefox_news","false"));
+				show_chrome = JSON.parse(XKit.tools.get_setting("xkit_show_chrome_news","false"));
+				show_safari = JSON.parse(XKit.tools.get_setting("xkit_show_safari_news","true"));
+			}
 			for (i=0;i<prev_objects.length;i++) {
-				console.log(prev_objects[i]);
-				if (prev_objects[i].read === false) {
-					if (typeof prev_objects[i].important !== "undefined") {
-						if (show_all === false && prev_objects[i].important !== "1") {
-							continue;
-						}
-					}
+				if (!prev_objects[i].read) {
+					if(prev_objects[i].deleted) { continue; }
+					if(prev_objects[i].channels.browsers.chrome === true && !show_chrome) { continue; }
```

Comment:
This sort of logic (if `channels.browsers.chrome && !show_chrome`) is duplicated around line 530, which is an indication that this should be extracted into its own function. Maybe something like:

```js
function shouldIgnore(news_item) {
    var show_chrome = JSON.parse(XKit.tools.get_setting("xkit_show_chrome_news", XKit.browser().chrome);
    var show_information = JSON.parse(XKit.tools.get_setting("xkit_show_information", "true"));
    var show_developer_news = JSON.parse(XKit.tools.get_setting("xkit_show_developers","true"));
    ...
    return news_item.channels.browsers.chrome && !show_chrome
        || news_item.channels.type === "information" && !show_information
        || news_item.channels.type === "developers" && !show_developer_news
        ...
}
```